qa/rpc-tests/wallet_addresses.py

```diff
@@ -19,7 +19,7 @@ def run_test(self):
         def addr_checks(default_type):
             # Check default type, as well as explicit types
             types_and_addresses = [
-                (default_type, self.nodes[0].z_getnewaddress()),
+                (default_type, self.nodes[0].z_getnewaddress('sprout')),
```

Comment:
I don't think this is the right change; I think you should leave this line unchanged but modify the five callers to this function (`addr_checks()`) to pass `sapling` instead of `sprout`, because the argument is exactly the default type (which is changing)..

Reply:
All the tests as they exist before this PR must explicitly ask for a sapling address, otherwise its the default sprout.  Therefore, any call to z_getnewaddress() intends to receive a sprout address.  Thats why I make the change here.  I also ran the full test suite locally and it passed.

Reply:
The comment is now wrong though, since this doesn't test the default. (There should be another test for the default.)

Reply:
Ah, got it now.  Thanks.  Fixed as @LarryRuane described.  Also filed https://github.com/zcash/zcash/issues/3686 for @daira.